subprojects/performance/src/performanceTest/groovy/org/gradle/performance/regression/corefeature/FileSystemWatchingPerformanceTest.groovy

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright 2020 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.performance.regression.corefeature
+
+import org.gradle.initialization.StartParameterBuildOptions
+import org.gradle.performance.AbstractCrossVersionPerformanceTest
+import org.gradle.performance.fixture.IncrementalAndroidTestProject
+import org.gradle.performance.fixture.IncrementalTestProject
+import org.gradle.performance.fixture.TestProjects
+
+class FileSystemWatchingPerformanceTest extends AbstractCrossVersionPerformanceTest {
+    private static final String AGP_TARGET_VERSION = "4.0"
+
+    def "assemble for non-abi change with file system watching"() {
+        IncrementalTestProject testProject = findTestProjectAndSetupRunnerForFsWatching()
+        testProject.configureForNonAbiChange(runner)
+
+        when:
+        def result = runner.run()
+        then:
+        result.assertCurrentVersionHasNotRegressed()
+    }
+
+    def "assemble for abi change with file system watching"() {
+        IncrementalTestProject testProject = findTestProjectAndSetupRunnerForFsWatching()
+
+        testProject.configureForAbiChange(runner)
+
+        when:
+        def result = runner.run()
+        then:
+        result.assertCurrentVersionHasNotRegressed()
+    }
+
+    IncrementalTestProject findTestProjectAndSetupRunnerForFsWatching() {
+        IncrementalTestProject testProject = TestProjects.projectFor(runner.testProject)
+        if (testProject instanceof IncrementalAndroidTestProject) {
+            testProject.configureForLatestAgpVersionOfMinor(runner, AGP_TARGET_VERSION)
```

Comment:
These methods should be static on `AndroidTestProject`.

Reply:
They are static now.

Reply:
They should be moved to `AndroidTestProject`. We can also do away with the `if` here.

Reply:
I think these methods only work for the Santa tracker projects and that is why they ended up in IncrementalAndroidTestProject. I wonder if it is too surprising to add them to AndroidTestProject and then they have no effect on the Android version.